tools/clang-format/clang-format-diff.py

```diff
@@ -114,15 +132,17 @@ def main():
       sys.exit(p.returncode)
 
     if not args.i:
-      with open(filename) as f:
-        code = f.readlines()
+      tmpfile.seek(0)
+      code = tmpfile.readlines()
       formatted_code = StringIO(stdout).readlines()
       diff = difflib.unified_diff(code, formatted_code,
-                                  filename, filename,
+                                  filename, tmpfile.name,
```

Comment:
I'm surprised that you are still using 'filename' here, isn't the (potentially modified) local copy not what you want ?

Reply:
Those two fields are only here to print the filenames, the diff being made from the `code` and `formatted_code` buffers. I left one `filename` so that the user knows which file it is
while still knowing that it is not the local file that is compared:

```
--- api/src/api/dpu.c	(before formatting)
+++ /tmp/tmpLPiTgF	(after formatting)
```

rather than having:

```
--- /tmp/tmpLPiTgF	(before formatting)
+++ /tmp/tmpLPiTgF	(after formatting)
```

Reply:
then you want both side to be filename ?